apps/api/src/reservation/api/http/inventory.controller.ts

```diff
@@ -18,44 +17,50 @@ import { Request } from 'express';
 import { ApiBearerAuth, ApiQuery, ApiTags } from '@nestjs/swagger';
 import { Permissions } from 'shared/auth/permission.decorator';
 import { AuthGuard } from 'shared/auth/auth.guard';
+import { GetInventoryItemsQuery } from 'reservation/application/queries/get-inventory-items.query';
+import { CommandBus, QueryBus } from '@nestjs/cqrs';
+import { UpsertLocationCommand } from 'reservation/application/commands/upsert-location.command';
+import { UpsertInventoryItemCommand } from 'reservation/application/commands/upsert-inventory-items.command';
 
 // TODO: Add DTOs for the request and response payloads
 @Controller('inventory-items')
 @ApiTags('inventory-items')
 @ApiBearerAuth('access-token')
 export class InventoryController {
-  constructor(private inventoryService: InventoryService) {}
+  constructor(
+    private readonly commandBus: CommandBus,
+    private readonly queryBus: QueryBus,
+  ) {}
 
   @Post()
   @UseGuards(WebhookSignatureGuard)
   async create(@Body() body: any) {
     if (body['_type'] === 'location') {
-      const dto = {
-        cmsId: body._id,
-        title: body.title,
-        address: body.address,
-      };
-      await this.inventoryService.upsertLocation(dto);
+      await this.commandBus.execute(
+        new UpsertLocationCommand(body._id, body.title, body.address),
+      );
     }
     if (body['_type'] === 'room') {
-      const dto = {
-        cmsId: body._id,
-        type: 'room',
-        title: body.title,
-        data: { capacity: body.capacity },
-        locationCmsId: body.location._ref,
-      };
-      await this.inventoryService.upsertInventoryItem(dto);
+      await this.commandBus.execute(
```

Comment:
Maybe it is better to use else if here, or early returns to prevent multiple if statement execution, although that is less likely